test/integration/spec/room.js

```diff
@@ -147,6 +147,74 @@ describe('Room', function() {
     });
   });
 
+  (defaults.topology === 'group' ? describe : describe.skip)('large rooms', () => {
+    let roomSid = null;
+    let aliceRoom = null;
+    let bobRoom = null;
+    let track = null;
+    let roomDetails = null;
+
+    before(async () => {
+      // create a large group room
+      roomSid = await createRoom(randomName(), defaults.topology, { MaxParticipants: 51 });
+      roomDetails = await getRoom(roomSid);
+
+      assert(roomDetails.max_concurrent_published_tracks === 16);
+      assert(roomDetails.max_participants === 51);
+      track = await createLocalAudioTrack({ fake: true });
+
+      // alice join the room
+      aliceRoom = await connect(getToken('Alice'), Object.assign({ name: roomSid }, defaults, { tracks: [] }));
+    });
+
+    after(async () => {
+      [aliceRoom, bobRoom].forEach(room => room && room.disconnect());
+      await completeRoom(roomSid);
+      track.stop();
+    });
+
+    it('does not send participantConnected when a participant joins with no tracks', async () => {
+      const bobConnectedPromise = new Promise(resolve => aliceRoom.on('participantConnected', resolve));
+      bobRoom = await connect(getToken('Bob'), Object.assign({ name: roomSid }, defaults, { tracks: [] }));
+
+      await waitForNot(bobConnectedPromise, `received unexpected participantConnected: ${roomSid}`);
+    });
+
+    it('emits participantConnected when a participant publishes track', async () => {
```

Comment:
@makarandp0 Can you add a test for the case where "participantConnected" is emitted for Bob when they join a Room with a track in ConnectOptions?

Reply:
sure, added a new case now.